### PR TITLE
Make usage of the AIO optional

### PR DESCRIPTION
**Pull request by pyakushin**

### CODE REVIEW

File: src/blob/file.rs [hunk 1]
```diff
@@ -112,7 +157,17 @@ impl File {
     }
 
     pub(crate) async fn fsyncdata(&self) -> IOResult<()> {
-        let compl = self.ioring.fdatasync(&*self.no_lock_fd);
-        compl.await
+        if let Some(ref ioring) = self.ioring {
+            let compl = ioring.fdatasync(&*self.no_lock_fd);
+            compl.await
+        } else {
+            let (tx, rx) = channel();
```
Comment (agend) [inline src/blob/file.rs:164]: I think could be as well as designated func

File: tests/common.rs [hunk 2]
```diff
@@ -71,15 +71,16 @@ pub async fn create_test_storage(
...
         .work_dir(&path)
         .blob_file_name_prefix("test")
         .max_blob_size(max_blob_size)
         .max_data_in_blob(100_000)
         .set_filter_config(Default::default())
-        .allow_duplicates();
-    let ioring = rio::new().expect("create uring");
-    let mut storage = builder.build(ioring).unwrap();
+        .allow_duplicates()
+        .enable_aio(ioring);
```
Comment (agend) [inline tests/common.rs:82]: Could also provide func to test for support of ioring.

Comment (pyakushin) [reply tests/common.rs:82]: ioring is created outside of pearl, so if user can't create rio, he won't be able to enable aio
```
let rio: Result<Rio, Error> = rio::new();
let storage = if let Ok(ioring) = rio {
    builder
        .enable_aio(ioring)
        .build()
} else {
    builder.build()
};
```

File: src/blob/file.rs (hunk 1 above)
Comment (pyakushin) [reply src/blob/file.rs]: Done

**Pull request merged**